### ISSUE TITLE
[BUG] Link props passed to SfButton shouldm't have default value ""

### ISSUE THREAD

**Issue by justyna-13** [vuestorefront/storefront-ui#2138]
**Describe the bug**
There are components (SfCallToAction, SfProductCard, SfProductCardHorizontal, SfCard, and more) that have `link` prop that is passed to SfButton. These link props have default value `""`. If there's no link passed then SfButton then under the hood is a button tag with a link attribute that isn't semantically correct. Such prop should have a default value of null

**How to reproduce**
1. Go to SfCallToAction (or another that has link prop passed to SfButton)
2. Leave link prop empty
2. See HTML
3. See button tag has `link` attribute

**Expected behavior**
Link props shouldn't have default value `""`

**Actual behavior**
Link props have default value `null`

**Comment by Ramiellll** [comment]
@justyna-13 Can you assign me this issue? I would love to help!

**Comment by justyna-13** [comment]
@Ramiellll Sure :)
